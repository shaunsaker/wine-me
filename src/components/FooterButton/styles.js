import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    paddingVertical: styleConstants.dimensions.spacing.vertical,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
    backgroundColor: styleConstants.colors.primary,
    flexDirection: 'row',
    alignItems: 'center',
    ...styleConstants.shadows.small,
  },
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.white,
    marginRight: styleConstants.dimensions.spacing.horizontal / 2,
  },
  text: {
    ...styleConstants.fonts.types.heading,
    color: styleConstants.colors.white,
    flex: 1,
    textAlign: 'center',
    marginRight: styleConstants.fonts.sizes.icon + styleConstants.dimensions.spacing.horizontal / 2, // align to center
  },
});

export default styles;
