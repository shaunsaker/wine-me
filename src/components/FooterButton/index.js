import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, ActivityIndicator } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styleConstants from '../../styleConstants';
import styles from './styles';

import Touchable from '../Touchable';

const propTypes = {
  handlePress: PropTypes.func,
  iconName: PropTypes.string,
  text: PropTypes.string,
  disabled: PropTypes.bool,
  alternateStyle: PropTypes.bool,
  showLoader: PropTypes.bool,
};

const defaultProps = {};

const FooterButton = ({ handlePress, iconName, text, disabled, alternateStyle, showLoader }) => {
  const mainComponent = showLoader ? (
    <ActivityIndicator size="small" color={styleConstants.colors.white} />
  ) : (
    <View style={styles.contentContainer}>
      <Icon name={iconName} style={[styles.icon, alternateStyle && styles.alternateText]} />

      <Text style={[styles.text, alternateStyle && styles.alternateText]}>{text}</Text>
    </View>
  );

  return (
    <Touchable
      onPress={handlePress}
      style={[styles.container, alternateStyle && styles.alternateContainer]}
      disabled={disabled}
    >
      {mainComponent}
    </Touchable>
  );
};

FooterButton.propTypes = propTypes;
FooterButton.defaultProps = defaultProps;

export default FooterButton;
