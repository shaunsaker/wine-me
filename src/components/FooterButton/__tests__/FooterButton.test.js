import React from 'react';
import renderer from 'react-test-renderer';

import FooterButton from '..';

describe('FooterButton', () => {
  const handlePress = jest.fn();
  const iconName = 'check';
  const text = 'Test';

  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(
        <FooterButton handlePress={handlePress} iconName={iconName} text={text} />,
      );

      expect(component).toMatchSnapshot();
    });

    it('renders the alternate style', () => {
      const component = renderer.create(
        <FooterButton handlePress={handlePress} iconName={iconName} text={text} alternateStyle />,
      );

      expect(component).toMatchSnapshot();
    });

    it('renders the loading state', () => {
      const component = renderer.create(
        <FooterButton handlePress={handlePress} iconName={iconName} text={text} showLoader />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
