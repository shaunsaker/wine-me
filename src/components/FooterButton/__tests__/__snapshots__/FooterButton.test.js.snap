// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterButton renders renders 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#E23E57",
      "borderColor": null,
      "borderWidth": 0,
      "elevation": 2,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 15,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.17,
      "shadowRadius": 2,
    }
  }
  testID="footerButton"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
              "marginRight": 7.5,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": undefined,
            "fontSize": 17,
            "fontWeight": "600",
            "marginRight": 31.5,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;

exports[`FooterButton renders renders the alternate style 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": null,
      "borderWidth": 0,
      "elevation": 2,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 15,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.17,
      "shadowRadius": 2,
    }
  }
  testID="footerButton"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
              "marginRight": 7.5,
            },
            Object {
              "color": "#E23E57",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": undefined,
            "fontSize": 17,
            "fontWeight": "600",
            "marginRight": 31.5,
            "textAlign": "center",
          },
          Object {
            "color": "#E23E57",
          },
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;

exports[`FooterButton renders renders the loading state 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#E23E57",
      "borderColor": null,
      "borderWidth": 0,
      "elevation": 2,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 15,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.17,
      "shadowRadius": 2,
    }
  }
  testID="footerButton"
>
  <ActivityIndicator
    animating={true}
    color="#FFFFFF"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
